### DISCUSSION TITLE
Provider registrering

### DISCUSSION THREAD

**Discussion by VladyslavParashchenko** [adonisjs/core#2139] [Help]
I created a new adonis package. It is client for memcached https://www.npmjs.com/package/adonis5-memcached-client 
But I got error when I tried to add this package to app. Unfortunately, I  got IoC container instance as constructor params to my provider. But I expected to get adonis app instance. 
Could you check my provider source code: https://github.com/VladyslavParashchenko/adonis5-memcached-client/blob/main/providers/AdonisMemcachedClientProvider.ts
My provider looks like https://github.com/adonisjs/lucid/blob/develop/providers/DatabaseProvider.ts lucid provider. But lucid provider gets expected params. I don't understand where is problem, in my code or with your IoC container.

**Comment by VladyslavParashchenko** [comment]
I fixed provider according to params that coming to constructor. Please, looking for problem version by this link https://github.com/VladyslavParashchenko/adonis5-memcached-client/blob/provider_error/providers/AdonisMemcachedClientProvider.ts

**Comment by VladyslavParashchenko** [comment]
@thetutlage could you help me ?

**Reply by thetutlage** [reply]
If you look at the Database provider, it has this `needsApplication` property set to true https://github.com/adonisjs/lucid/blob/develop/providers/DatabaseProvider.ts#L17. You will have to do the same

**Reply by VladyslavParashchenko** [reply]
> If you look at the Database provider, it has this `needsApplication` property set to true https://github.com/adonisjs/lucid/blob/develop/providers/DatabaseProvider.ts#L17. You will have to do the same

I understand, thank you for response.
